src/st/dre/11A64.c

```diff
@@ -134,79 +448,459 @@ s16 func_8019A718(void) {
     return var_a0;
 }
 
-INCLUDE_ASM("asm/st/dre/nonmatchings/11A64", func_8019A75C);
+void func_8019A75C(void) { // MoveEntity
+    D_8006C3B8->posX.value += D_8006C3B8->accelerationX;
+    D_8006C3B8->posY.value += D_8006C3B8->accelerationY;
+}
 
-INCLUDE_ASM("asm/st/dre/nonmatchings/11A64", func_8019A78C);
+void func_8019A78C(void) { // FallEntity
+    if (D_8006C3B8->accelerationY < FALL_TERMINAL_VELOCITY) {
+        D_8006C3B8->accelerationY += FALL_GRAVITY;
+    }
+}
 
 INCLUDE_ASM("asm/st/dre/nonmatchings/11A64", func_8019A7B8);
 
 INCLUDE_ASM("asm/st/dre/nonmatchings/11A64", func_8019AA30);
 
-INCLUDE_ASM("asm/st/dre/nonmatchings/11A64", func_8019AC18);
+Entity* func_8019AC18(Entity* start, Entity* end) {
+    Entity* current = start;
+
+    while (current < end) {
+        if (current->objectId == 0) {
+            func_8019A3A8(current);
+            return current;
+        }
+
+        current++;
+    }
+    return NULL;
+}
 
+// This function matches with PSYQ4.0 GCC 2.7.2 with -01 and -02 Optimization
+// at -v0 reg swap
+// flags https://decomp.me/scratch/dlcph
+#ifndef NON_MATCHING
 INCLUDE_ASM("asm/st/dre/nonmatchings/11A64", func_8019AC78);
+#else
+extern s16 D_80180D80[];
+
+s32 func_8019AC78(u8 arg0, s16 arg1) { return D_80180D80[arg0] * arg1; }
```

Comment:
Pretty sure if you merge `master` to this one, you can have a match here and at `func_8019ACA4` thanks to the recent implementation of `ASPATCH`!